rio_tiler/io/stac.py

```diff
@@ -47,10 +48,10 @@ def fetch(filepath: str) -> Dict:
     if parsed.scheme == "s3":
         bucket = parsed.netloc
         key = parsed.path.strip("/")
-        return json.loads(aws_get_object(bucket, key))
+        return json.loads(aws_get_object(bucket, key, **kwargs))
 
     elif parsed.scheme in ["https", "http", "ftp"]:
-        return requests.get(filepath).json()
+        return requests.get(filepath, **kwargs).json()
```

Comment:
Seems like a possible issue with this is that a user can't use a single `fetch_options` for data that _sometimes_ has HTTPS urls and _sometimes_ has `s3:` urls... 

I.e. if you have a `params` key in your `fetch_options` but your data unexpectedly has an asset with an `s3://` url, then `params` will be passed to `aws_get_object` and fail, right?

Reply:
fetch_options is not used at the asset level but just to get the `Items`

Reply:
A solution we've used for problems like this in loaders.gl is to have nested options keys, so for example `fetch_options` would take options that look like
```js
{
	'https': {
		params: {}
	},
	's3': {
		request_pays: True,
	}
}
```
where the `https` key gets passed to `requests` and the `s3` key gets passed to `aws_get_object`.

Reply:
> fetch_options is not used at the asset level but just to get the `Items`

oh, nevermind then. If it's just to get the STAC Item itself, it shouldn't be too hard for the end user to know where their stac data is coming from